tests/test_bdd.py

```diff
@@ -37,3 +47,70 @@ def check_entry_list(homepage):
     html = homepage.html
     entries = html.find_all('article', class_='entry')
     assert len(entries) == 3
+
+
+@scenario('features/homepage.feature', 'Markdown')
+def test_markdown():
+    pass
+
+
+@scenario('features/homepage.feature', 'Colorized Code')
+def test_color():
+    pass
+
+
+@given('an author')
+def check_authorization(homepage, request):
+    manager = BCRYPTPasswordManager()
+    settings = {
+        'auth.username': 'admin',
+        'auth.password': manager.encode('secret'),
+    }
+    testing.setUp(settings=settings)
+    req = testing.DummyRequest()
+
+    def cleanup():
+        testing.tearDown()
+
+    request.addfinalizer(cleanup)
+
+    return req
+
+
+@when('the user adds an entry')
+def test_entry(db_session):
+    data = {u'text': u'#test text\n**bold**```print "Hello World"'
+            '```', u'title': u"Title"}
+    journal.Entry.write(session=db_session, **data)
+    db_session.flush()
+
+
+@then('that entry fomats with markdown')
+def format_markdown(db_session, app):
+    response = app.get('/')
+    actual = response.body
+    expected = u'<h1>test text</h1>'
+    bold = u'<strong>bold</strong>'
+    assert expected in actual
+    assert bold in actual
```

Comment:
I'd rather see you use the `.html` attribute of the response to get a beautiful soup object and then use the `find` or `find_all` apis of that object to get at the DOM nodes you are looking for here.